12-offer-encoding.md

```diff
@@ -0,0 +1,1019 @@
+    1. type: 48 (`fallbacks`)
+    2. data:
+        * [`u8`:`num`]
+        * [`num*fallback_address`:`fallbacks`]
+    1. type: 52 (`refund_signature`)
+    2. data:
+        * [`bip340sig`:`payer_signature`]
+    1. type: 56 (`replace_invoice`)
+    2. data:
+        * [`sha256`:`payment_hash`]
+    1. type: 240 (`signature`)
+    2. data:
+        * [`bip340sig`:`sig`]
+
+1. subtype: `blinded_payinfo`
+2. data:
+   * [`u32`:`fee_base_msat`]
+   * [`u32`:`fee_proportional_millionths`]
+   * [`u16`:`cltv_expiry_delta`]
+   * [`u16`:`flen`]
+   * [`flen*byte`:`features`]
+
+1. subtype: `fallback_address`
+2. data:
+   * [`byte`:`version`]
+   * [`u16`:`len`]
+   * [`len*byte`:`address`]
+
+## Requirements
+
+A writer of an invoice:
+  - MUST set `timestamp` to the number of seconds since Midnight 1
+    January 1970, UTC.
+  - MUST set `payment_hash` to the SHA2 256-bit hash of the
+    `payment_preimage` that will be given in return for payment.
+  - MUST set (or not set) `send_invoice` the same as the offer.
+  - MUST set `offer_id` to the id of the offer.
+  - MUST specify exactly one signature TLV: `signature`.
+    - MUST set `sig` to the signature using `node_id` as described in [Signature Calculation](#signature-calculation).
+  - if the chain for the invoice is not solely bitcoin:
+    - MUST specify `chains` the invoice is valid for.
+  - otherwise:
+    - the bitcoin chain is implied as the first and only entry.
+  - if it has bolt11 features:
+    - MUST set `features` to the bitmap of features.
+  - if the invoice corresponds to an offer with `recurrence`:
+    - MUST set `recurrence_basetime` to the start of period #0 as calculated
+      by [Period Calculation](#offer-period-calculation).
+    - if it sets `relative_expiry`:
+      - MUST NOT set `relative_expiry` `seconds_from_timestamp` more than the number of seconds after `timestamp` that payment for this period will be accepted.
+  - otherwise:
+    - MUST not set `recurrence_basetime`.
+  - if the expiry for accepting payment is not 7200 seconds after `timestamp`:
+    - MUST set `relative_expiry` `seconds_from_timestamp` to the number of
+      seconds after `timestamp` that payment of this invoice should not be attempted.
+  - if the `min_final_cltv_expiry` for the last HTLC in the route is not 18:
+    - MUST set `min_final_cltv_expiry`.
+  - if it accepts onchain payments:
+    - MAY specify `fallbacks`
+    - MUST specify `fallbacks` in order of most-preferred to least-preferred
+      if it has a preference.
+    - for the bitcoin chain, it MUST set each `fallback_address` with
+      `version` as a valid witness version and `address` as a valid witness
+      program
+  - if it is connected only by private channels:
+    - MUST include a `blinded_path` containing one or more paths to the node.
+  - otherwise:
+    - MAY include `blinded_path`.
+  - if it includes `blinded_path`:
+    - MUST specify `path` in order of most-preferred to least-preferred if
```

Comment:
what is the metric for this? the local inbound capacity of the recipient? 

Also in the case of larger payments and MPP the sender might need to bind to several blinded paths that are in the hints. I think it is worthwhile considering to add a more specific routing hint of saying how many sats one can accept on that blinded path (I understand that the recipient cannot guarantee the liquidity on the path but I think the hints could be understood as an upper bound)